user/deployment-v2.md

```diff
@@ -0,0 +1,207 @@
+support for a particular service, the respective provider is marked with one of
+the following maturity levels, according to the given criteria:
+
+* `dev` - the provider is in development (initial level)
+* `alpha` - the provider is fully tested
+* `beta` - the provider has been in alpha for at least a month and successful real-world production deployments have been observed
+* `stable` - the provider has been in beta for at least two months and there are no open issues that qualify as critical (such as deployments failing, documented functionality broken, etc.)
+
+Dpl v2 represents a major rewrite, so support for all providers has been
+reset to `dev` or `alpha`, depending on the test status.
+
+For all levels except `stable` a message will be printed to your build log
+that informs you about the current status.
+
+## Cleaning up the Git Working Directory
+
+The previous version of dpl - our deployment integration tooling - used to
+reset your working directory and delete all changes made during the build
+using `git stash --all`. In order to keep changes one had to opt out using
+`skip_cleanup: true`. This default turned out to be useful only for very few
+providers and has been changed in dpl v2.
+
+If you do need to clean up the working directory from any changes made during
+the build process, please opt in to cleanup by adding the following to your
+`.travis.yml` file:
+
+```yaml
+deploy:
+  cleanup: true
+```
+{: data-file=".travis.yml"}
+
+## Deploying to Multiple Providers
+
+Deploying to multiple providers is possible by adding the different providers
+to the `deploy` section as a list. For example, if you want to deploy to both
+cloudControl and Heroku, your `deploy` section would look something like this:
+
+```yaml
+deploy:
+  - provider: cloudcontrol
+    email: "YOUR CLOUDCONTROL EMAIL"
+    password: "YOUR CLOUDCONTROL PASSWORD"
+    deployment: "APP_NAME/DEP_NAME"
+  - provider: heroku
+    api_key: "YOUR HEROKU API KEY"
+```
+{: data-file=".travis.yml"}
+
+## Conditional Releases with `on:`
+
+Set your build to deploy only in specific circumstances by configuring the `on:` key for any deployment provider.
+
+```yaml
+deploy:
+  provider: s3
+  access_key_id: "YOUR AWS ACCESS KEY"
+  secret_access_key: "YOUR AWS SECRET KEY"
+  bucket: "S3 Bucket"
+  skip_cleanup: true
+  on:
+    branch: release
+    condition: $MY_ENV = super_awesome
+```
+{: data-file=".travis.yml"}
+
+When *all* conditions specified in the `on:` section are met, your build will deploy.
+
+Use the following options to configure conditional deployment:
+
+* `repo`: in the form `owner_name/repo_name`. Deploy only when the build occurs on a particular repository. For example
+
+   ```yaml
+   deploy:
+     provider: s3
+     on:
+       repo: travis-ci/dpl
+   ```
+   {: data-file=".travis.yml"}
+
+* `branch`: name of the branch.
+   If omitted, this defaults to the `app`-specific branch or `master`. If the branch name is not known ahead of time, you can specify
```

Comment:
```suggestion
   If omitted, this defaults to the `app`-specific branch or `master`. If the branch name is not known ahead of time, you can specify
```

what does the `app`-specific branch mean? The branch that the specific deployment provider would look for by default?

Reply:
I don't know ... I think I've just copied that text from the existing page?